test/client/api/directoryAdmin.unittest.ts

```diff
@@ -0,0 +1,326 @@
+            directory.storedApplications.add(fakeApp4);
+        }, createPopulatedDirectoryWithStoredApplications([fakeApp4, fakeApp5, fakeApp6])],
+
+        ['We can attempt to remove a stored application not in the app directory and the directory will not modified', (directory: AppDirectory) => {
+            directory.storedApplications.remove(fakeApp1);
+        }, populatedDirectory]
+    ];
+
+    test.each(testParams)('%s', async (
+        titleParam: string,
+        migrationHandler: UpdateAppDirectoryMigrationHandler,
+        expectedAppDirectory: StoredDirectoryShardMap
+    ) => {
+        await updateAppDirectory(migrationHandler, {namespace: 'namespace-2'});
+
+        expect(fin.Storage.setItem).toBeCalledWith(APP_DIRECTORY_STORAGE_TAG, JSON.stringify(expectedAppDirectory));
+    });
+
+    test('We can create an app directory in a new namespace', async () => {
+        await updateAppDirectory(() => {}, {
+            namespace: 'namespace-3'
+        });
+
+        expect(fin.Storage.setItem).toBeCalledWith(APP_DIRECTORY_STORAGE_TAG, JSON.stringify({
+            ...populatedDirectory,
+            'namespace-namespace-3': {
+                version: 1,
+                remoteSnippets: [],
+                storedApplications: []
+            }
+        }));
+    });
+
+    test('We can update the version of the app directory', async () => {
+        await updateAppDirectory(() => {}, {
+            namespace: 'namespace-2',
+            destinationVersion: populatedDirectoryVersion + 1
+        });
+
+        expect(fin.Storage.setItem).toBeCalledWith(APP_DIRECTORY_STORAGE_TAG, JSON.stringify({
+            ...populatedDirectory,
+            'namespace-namespace-2': {
+                ...populatedDirectory['namespace-namespace-2'],
+                version: populatedDirectoryVersion + 1
+            }
+        }));
+    });
+
+    test('If the specificed version is lower the than version of the app directory, the migration function is not called \
+and the directory is not modified', async () => {
+        const spy = jest.fn<void, [AppDirectory]>();
+
+        await updateAppDirectory(spy, {
+            namespace: 'namespace-2',
+            destinationVersion: populatedDirectoryVersion - 1
+        });
+
+        expect(spy).not.toBeCalled();
+        expect(fin.Storage.setItem).not.toBeCalled();
+    });
+
+    test('If the specificed max version is lower the than version of the app directory, the migration function is not called \
+and the directory is not modified', async () => {
+        const spy = jest.fn<void, [AppDirectory]>();
+
+        await updateAppDirectory(spy, {
+            namespace: 'namespace-2',
+            maxSourceVersion: populatedDirectoryVersion - 1
+        });
+
+        expect(spy).not.toBeCalled();
+        expect(fin.Storage.setItem).not.toBeCalled();
+    });
+
+    test('If the specificed max version is equal to version of the app directory, the migration function is called \
```

Comment:
typo: specificed -> specified

Reply:
there are 3 more of this typo in this file.

Reply:
✅

✅✅✅